File: frontend/sirooo_ai/lib/models/schedule.dart
class Schedule {
  final int? id;
  final String name;
  final DateTime date;
  final String startTime;
  final String endTime;
  final String location;
  final String eventType;

  Schedule({
    this.id,
    required this.name,
    required this.date,
    required this.startTime,
    required this.endTime,
    required this.location,
    required this.eventType,
  });

  factory Schedule.fromJson(Map<String, dynamic> json) {
    return Schedule(
      id: json['id'],
      name: json['name'],
      date: DateTime.parse(json['date']),
      startTime: json['start_time'],
      endTime: json['end_time'],
      location: json['location'],
      eventType: json['event_type'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'name': name,
      'date': date.toIso8601String().split('T')[0],
      'start_time': startTime,
      'end_time': endTime,
      'location': location,
      'event_type': eventType,
    };
  }
}

File: backend/app/schemas/schedule.py
from pydantic import BaseModel
import datetime
from typing import Optional

class ScheduleBase(BaseModel):
    name: str
    date: datetime.date
    start_time: datetime.time
    end_time: datetime.time
    location: str
    event_type: str

class ScheduleCreate(ScheduleBase):
    pass

class ScheduleUpdate(BaseModel):
    name: str | None = None
    date: datetime.date | None = None
    start_time: datetime.time | None = None
    end_time: datetime.time | None = None
    location: str | None = None
    event_type: str | None = None

class ScheduleResponse(ScheduleBase):
    id: int

    class Config:
        from_attributes = True
